staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +51,37 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ParallelizeUntil is a framework that allows for parallelizing N
+// independent pieces of work until the context done or canceled.
+func ParallelizeUntil(ctx context.Context, workers, pieces int, doWorkPiece DoWorkPieceFunc) {
+	// If the passed ctx is nil, then the default context.TODO() is used.
+	if ctx == nil {
+		ctx = context.TODO()
```

Comment:
maybe change to `return Parallelize(workers, pieces, doWorkPiece)` so that we don't need to check if context is done in `select{}` for this "ctx == nil" case

Reply:
Per other comments, I think the plan is for `Parallelize` to call this func with a nil context to eliminate duplicate code.

`context.TODO()` implies that we need further code changes to get an appropriate context. Since your goal is to create a context that never cancels, it is more appropriate to use `context.Background()`.

However, I think it would be cleaner to write it as,
```golang
var stop chan struct{}
if ctx != nil {
    stop = ctx.Done()
}
```

Reply:
This is not a good idea, ParallelizeUntil should not rely on Parallelize.Because ParallelizeUntil may replace Parallelize.

Reply:
That's fair.